tendrl/node_agent/discovery/sds/plugins/discover_ceph_sds_details.py

```diff
@@ -25,26 +26,15 @@ def discover_storage_system(self):
             return ret_val
 
         if out:
-            details = json.loads(out)
+            details = out.split()
 
-            ret_val['pkg_version'] = details['version'].split()[2]
-            ret_val['pkg_name'] = details['version'].split()[0]
+            ret_val['pkg_version'] = details[2]
+            ret_val['pkg_name'] = details[0]
 
             # get the cluster_id details
-            cmd = subprocess.Popen(
-                "ceph -s -f json",
-                shell=True,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE
-            )
-            out, err = cmd.communicate()
-            if err:
-                LOG.error("Error getting cluster details")
-                return ret_val
-            if out:
-                details = json.loads(out)
-                ret_val['detected_cluster_id'] = details['fsid']
-                ret_val['cluster_attrs'] = {'fsid': details['fsid'],
-                                            'name': 'ceph'}
+            raw_data = ini2json.ini_to_dict("/etc/ceph/ceph.conf")
```

Comment:
This will not work if you have a custom cluster name. In that case the config file name is something like <cluster-name>.conf